contracts/mainnet/connectors/aave/v2/main.sol

```diff
@@ -150,7 +150,11 @@ abstract contract AaveResolver is Events, Helpers {
 
         TokenInterface tokenContract = TokenInterface(_token);
 
-        _amt = _amt == uint(-1) ? getPaybackBalance(_token, rateMode) : _amt;
+        if (_amt == uint(-1)) {
+            uint _amtDSA = tokenContract.balanceOf(address(this));
```

Comment:
@shriyatyagii if token address is eth, then we need to check eth balance

Reply:
Updated - [here](https://github.com/Instadapp/dsa-connectors/pull/214/commits/fb9abb204216aab8a1fae7350cac952032f65b58)